nixos/modules/security/pam.nix

```
@@ -460,6 +460,94 @@ let
         description = lib.mdDoc "Contents of the PAM service file.";
       };
 
+      types = mkOption {
+        # This option is experimental and subject to breaking changes without notice.
+        visible = false;
+
+        description = lib.mdDoc ''
+          PAM rules for this service.
+
+          Attribute keys are the `type` of the rules (one of `account`, `auth`, `password`, `session`).
+        '';
+        type = types.attrsOf (types.submodule {
+          options.rules = mkOption {
+            description = lib.mdDoc ''
+              PAM rules for this service and type.
+
+              Attribute keys are the name of each rule.
+            '';
+            type = types.attrsOf (types.submodule ({ config, ... }: {
+              options = {
+                enable = mkOption {
+                  type = types.bool;
+                  default = true;
+                  description = lib.mdDoc ''
+                    Whether this rule is added to the PAM service config file.
+                  '';
+                };
+                order = mkOption {
+                  type = types.int;
```

Here's an idea to avoid the problem of not reserving enough space between rules :)
```suggestion
                  type = types.float;
```

---

:laughing: That is clever, but I prefer our security infrastructure not rely on floating point comparisons. I will increase the integer spacing. Even if someone needs to add hundreds of rules (unlikely!) they can use `include`/`substack` controls.

---

Increased spacing to 100 in ee5dd1dd40fa77339898580bb614db920b5af46d.